src/ev-matrix.c

```diff
@@ -175,33 +175,43 @@ on_matrix_open (GObject *object, GAsyncResult *result, gpointer user_data)
      * a sync callback for it as it will otherwise assert() */
     if (g_strcmp0 (cm_account_get_login_id (a), username) == 0) {
       client = g_steal_pointer (&c);
-      break;
+    } else {
+      cm_client_set_enabled (c, FALSE);
```

Comment:
```suggestion
      /* FIXME: libcmatrix shouldn't enable clients by default */
      cm_client_set_enabled (c, FALSE);
```

Reply:
I don't think libcmatrix does (but I may be wrong here). Otherwise I wouldn't have had to explicitly `cm_client_set_enabled (client, TRUE)`  below for the newly created account to login.

Instead this just ensures that we're only ever enabling a single one.

Reply:
I've not included this change, but reworded the commit message to make the current situation hopefully clearer.

Reply:
But we break out of the loop, how would we enable multiple ones?

Reply:
> But we break out of the loop

Not anymore? `break` has been removed

> how would we enable multiple ones?

by enabling the ones we want later on (as we do now for the single client/account).

I briefly considered multiple accounts, but since the code in the beginning only looks for a single account `matrix-00` I thought this would be a problem for the future.

Reply:
Again myself explaining things badly, sorry:

> Not anymore? break has been removed

Yes, my question is: why not just use the `break` as before. That would only ever enable a single account.

>  how would we enable multiple ones?

(that was related to: if we `break` out of the loop how can we ever end up enabling an account accidentally. I am basically trying to understand how you think we could get into a situation that requires `cm_client_set_enabled (c, FALSE);` if we `break` out of the loop after enabling the first account?

Reply:
I've not checked in detail, but I think`cm_client_set_enabled ()` persists as it is part of the stored secret.

So if we switch the account (by changing the username/pw of `matrix-00` in our config) I would assume (untested as of yet) that the old account would stay enabled which should then lead to the failed assertion of a sync callback not set.

Instead of the theory crafting it would probably be wise to confirm that this is actually what would happen,
but this will have to wait until tomorrow.

Reply:
Which circles us back to my initial whish for a `FIXME`: libcmatrix shouldn't enable clients by default :smile: - it's fine if it keeps the `enabled` state tracked in the secret store but it shouldn't start *enabling* when creating a matrix client object (blowing up if there's no callback connected, etc). That should be an explicit operation like:

- create `CmMatrix`
- get a list of clients for this application / configured accounts
- allow the app to check which should be enabled and then call `set_enabled` on them to actually start processing

For the simple use case we could add a `auto-enable` property to `CmMatrix` to keep the current behavior but I'd argue that's rarely what an app wants.

So from a code / commit message point of view this looks fine to me. It would just be good to add the `FIXME`.